aten/src/ATen/core/ivalue_inl.h

```diff
@@ -266,14 +270,22 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
   }
 
   // Check if the current future has completed
-  bool completed() {
+  bool completed() const{
     return completed_;
   }
 
   CAFFE2_API friend std::ostream& operator<<(
       std::ostream& out,
       const Future& v);
 
+  TypePtr type() {
+    if (!completed()) {
```

Comment:
This seems a little strange—should the type of a `Future` change depending on its completion state?